upgraded erlang18 to v18.3.4.11 and build function update

timin opened the pull request:
1. upgraded erlang18 to v18.3.4.11
2. updated build function as per change in source tarball
3. using step down autoconf v2.69 due to incompatibility with autoconf v2.71

Signed-off-by: Nimit <nimit.jyotiana@hotmail.com>

timin committed:
upgraded erlang18 to v18.3.4.11 and build function update

1. upgraded erlang18 to v18.3.4.11
2. updated build function as per change in source tarball
3. using step down autoconf v2.69 due to incompatibility with autoconf v2.71

Signed-off-by: Nimit <nimit.jyotiana@hotmail.com>

timin commented:
fixes issue https://github.com/habitat-sh/core-plans/issues/4125

rahulgoel1 commented on plan.sh:
```diff
@@ -1,16 +1,24 @@
 pkg_license=('Apache-2.0')
 pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
-pkg_source=http://www.erlang.org/download/otp_src_${pkg_version}.tar.gz
+pkg_source=https://api.github.com/repos/erlang/otp/tarball/refs/tags/OTP-${pkg_version}
```
Why are we changing the package source?

timin replied:
webpage for erlang v18.3.4.11 (https://www.erlang.org/patches/otp-18.3.4.11) suggest download link as mentioned in pkg_source. https://www.erlang.org/download does not have v18.3.4.11 available for download.

rahulgoel1 commented on plan.sh:
```diff
@@ -1,16 +1,24 @@
+       	core/openssl
+       	core/perl
+       	core/m4
+	core/autoconf/2.69
```
Need to discuss deps change and pinned version . Why did we change the deps and pin it to 2.69 for autoconf ?

mwrock replied:
I think we will need to pin to 2.69. All solutions I found on google were exactly that: use 2.69. Because this is a build dep of erlang and because autoconf is in almost all cases only going to be depended on as a build dep, this should not cause version conflicts. It would only be an issue when building erlang AND if one of erlangs other build/runtime deps had a runtime dep on autoconf (very unlikely). So if this did blow up, it would only do so while building erlang.

timin committed:
fix for build error due to unpack of source tarball

source tarball is untarred with different name with which is different from standard naming for processing of package building

Signed-off-by: Nimit <nimit.jyotiana@hotmail.com>
